sktime/transformations/series/peak.py

```diff
@@ -0,0 +1,441 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract peak/working hour features from datetimeindex."""
+__author__ = ["ali-parizad", "VyomkeshVyas"]
+__all__ = ["PeakTimeFeature"]
+
+
+import pandas as pd
+
+from sktime.transformations.base import BaseTransformer
+
+
+class PeakTimeFeature(BaseTransformer):
+    """PeakTime feature extraction for use in e.g. tree based models.
+
+    PeakTimeFeature uses a datetime index column and generates peak/working features
+    for e.g. peak hours, peak weak, peak month, working hours, etc. It works based on
+    the input intervals, e.g., peak_hour_start=[6], peak_hour_end=[9]
+
+    Parameters
+    ----------
+    ts_freq : str,
+        Restricts selection of items to those with a frequency lower than
+        the frequency of the time series given by ts_freq.
+        E.g., if daily data is provided and ts_freq = ("D"), it does not make
+        sense to derive PeakTimeFeature with higher frequency like hourly features. So,
+        the outpul will be is_peak_day, is_peak_week, is_peak_month, is_peak_quarter,
+        is_peak_year. Only supports the following frequencies:
+        * Y - year
+        * Q - quarter
+        * M - month
+        * W - week
+        * D - day
+        * H - hour
+
+    peak_[*]_start, peak_[*]_end : list,
+        peak start and peak end should be specified in form of
+        peak_[*]_start = [start1, start2, ...], peak_[*]_end = [end1, end2, ...].
+        E.g. 1, one peak interval: peak_hour_start=[6], peak_hour_end=[9] means we
+        have just ONW peak hour interval where peak starts at 6am and peak ends at 9am.
+        E.g. 2, two peak intervals: peak_hour_start=[6, 16], peak_hour_end=[9, 20]
+        means we have TWO peak hour intervals where the first peak starts at 6 am
+        and ends at 9 am. The second peak starts at 16 am and ends at 20.
+        - we may have more than TWO intervals.
+        The * can be one of the following:
+        * Y - year
+        * Q - quarter
+        * M - month
+        * W - week
+        * D - day
+        * H - hour
```

Comment:
Same as above

Reply:
Thanks for your suggestion! Used numpydoc to correct the style of doctoring. I did include all of these Peak hour, Peak day, ... separately, instead of using * and ^.